tests/test_cli.py

```diff
@@ -61,12 +64,179 @@ def mocked_command(command, args):
     assert "" == err
 
 
-@pytest.mark.parametrize("command", COMMANDS)
-def test_cli_verbose(command, monkeypatch):
-    def mocked_action(*args, **kwargs):
+@pytest.mark.parametrize("command", ENVIRONMENT_COMMANDS)
+def test_cli_verbose_env(command, monkeypatch, project_directory_factory):
+    if command == "create":
+        project_path = project_directory_factory()
+    else:
+        env_yaml = "dependencies: []\n"
+        project_path = project_directory_factory(env_yaml=env_yaml)
+
+    def mocked_action(*_, **kwargs):
+        assert kwargs.get("verbose", False)
+
+    monkeypatch.setattr(f"conda_project.project.Environment.{command}", mocked_action)
+
+    ret = parse_and_run([command, "--directory", str(project_path)])
+    assert ret == 0
+
+
+@pytest.mark.parametrize("command", PROJECT_COMMANDS)
+def test_cli_verbose_project(command, monkeypatch, project_directory_factory):
+    if command == "create":
+        project_path = project_directory_factory()
+    else:
+        env_yaml = "dependencies: []\n"
+        project_path = project_directory_factory(env_yaml=env_yaml)
+
+    def mocked_action(*_, **kwargs):
         assert kwargs.get("verbose", False)
 
     monkeypatch.setattr(f"conda_project.project.CondaProject.{command}", mocked_action)
 
-    ret = parse_and_run([command])
+    _ = parse_and_run([command, "--directory", str(project_path)])
+
+
+def test_create_with_prepare(tmpdir):
+    ret = parse_and_run(["create", "--directory", str(tmpdir), "--prepare"])
+
+    assert ret == 0
+
+    assert os.path.exists(
+        os.path.join(tmpdir, "envs", "default", "conda-meta", "history")
+    )
+
+
+@pytest.mark.parametrize("command", ENVIRONMENT_COMMANDS)
+def test_command_with_environment_name(command, monkeypatch, project_directory_factory):
+    env1 = env2 = "dependencies: []\n"
+    project_yaml = f"""\
```

Comment:
I don't want to harp on this more, but this is my new favorite way to format things like this I discovered this weekend: https://github.com/mattkram/rusty-mooring/blob/main/tests/test_config.py#L15

Just adding FYI.

Reply:
ok